internal/cloud/azure/azure.go

```diff
@@ -157,40 +159,34 @@ func (c *Cloud) GetCCMConfig(ctx context.Context, providerID string, cloudServic
 // The returned string is an IP address without a port, but the method name needs to satisfy the
 // metadata interface.
 func (c *Cloud) GetLoadBalancerEndpoint(ctx context.Context) (string, error) {
-	resourceGroup, err := c.imds.resourceGroup(ctx)
-	if err != nil {
-		return "", fmt.Errorf("retrieving resource group: %w", err)
-	}
-	uid, err := c.imds.uid(ctx)
-	if err != nil {
-		return "", fmt.Errorf("retrieving instance UID: %w", err)
-	}
+	return c.getLoadBalancerPublicIP(ctx)
+}
 
-	lb, err := c.getLoadBalancer(ctx, resourceGroup, uid)
+// GetAPIServerCertSANs retrieves the SANs of the apiserver certificate.
+// On Azure, the SANs are the load balancer IP, the wildcard DNS name
+// of public IP addresses in the chosen location and the (optional) custom
+// endpoint (if set).
+func (c *Cloud) GetAPIServerCertSANs(ctx context.Context) ([]string, error) {
+	publicIP, err := c.getLoadBalancerPublicIP(ctx)
 	if err != nil {
-		return "", fmt.Errorf("retrieving load balancer: %w", err)
-	}
-	if lb == nil || lb.Properties == nil {
-		return "", errors.New("could not dereference load balancer IP configuration")
-	}
-
-	var pubIP string
-	for _, fipConf := range lb.Properties.FrontendIPConfigurations {
-		if fipConf == nil || fipConf.Properties == nil || fipConf.Properties.PublicIPAddress == nil || fipConf.Properties.PublicIPAddress.ID == nil {
-			continue
-		}
-		pubIP = path.Base(*fipConf.Properties.PublicIPAddress.ID)
-		break
+		return nil, fmt.Errorf("retrieving public IP: %w", err)
 	}
-
-	resp, err := c.pubIPAPI.Get(ctx, resourceGroup, pubIP, nil)
+	dnsWildcard, err := c.getLoadBalancerDNSWildcard(ctx)
 	if err != nil {
-		return "", fmt.Errorf("retrieving load balancer public IP address: %w", err)
+		return nil, fmt.Errorf("retrieving DNS wildcard: %w", err)
 	}
-	if resp.Properties == nil || resp.Properties.IPAddress == nil {
-		return "", fmt.Errorf("could not resolve public IP address reference for load balancer")
+	customEndpoint, err := c.getCustomEndpoint(ctx)
+	var hasCustomEndpoint bool
+	if err == nil && len(customEndpoint) > 0 {
+		hasCustomEndpoint = true
+	} else {
+		log.Printf("No custom endpoint found: %v", err)
```

Comment:
```suggestion
```
Usage of `log` package.
Consider using our logging package, or remove the log statement

Reply:
Thanks! Good find. This was a left over thing from debugging.